src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -100,5 +132,33 @@ public void AddTagHelperAttribute([NotNull] string name, object value)
         {
             AllAttributes.Add(name, value);
         }
+
+        /// <summary>
+        /// Executes the child content asynchronously.
+        /// </summary>
+        /// <returns>A task that indicates when child content execution has completed.</returns>
+        public Task ExecuteChildContentAsync()
+        {
+            return _executeChildContentAsync();
+        }
+
+        /// <summary>
+        /// Execute and retrieve the rendered child content asynchronously.
+        /// </summary>
+        /// <returns>The rendered child content.</returns>
+        /// <remarks>
+        /// Child content is only executed once. Successive calls to this method return a cached result.
```

Comment:
nit: might need to clarify this remark since "cached" doesn't indicate the (limited) scope.  even possible for readers to wonder whether just the task`Task` (not its returned `string`) is preserved.  Perhaps `Child content is only executed once per view execution.  Successive calls to this method or successive executions of the returned <see cref="Task"/> ...` ?